File: algorithms/easy/construct_the_minimum_bitwise_array_i.rs
impl Solution {
    pub fn min_bitwise_array(nums: Vec<i32>) -> Vec<i32> {
        let mut arr = Vec::new();
        for &n in &nums {
            if n == 2 {
                arr.push(-1);
            }
            for m in 1..=n {
                if m | (m + 1) == n {
                    arr.push(m);
                    break;
                }
            }
        }
        arr
    }
}

File: algorithms/easy/construct_the_minimum_bitwise_array_i.kt
class Solution {
  fun minBitwiseArray(nums: List<Int>): IntArray {
    val ans = IntArray(nums.size)
    for ((i, n) in nums.withIndex()) {
      if (n % 2 == 0) {
        ans[i] = -1
        continue
      }

      var m = 1
      while (m or (m + 1) != n) {
        m += 1
      }
      ans[i] = m
    }

    return ans
  }
}
